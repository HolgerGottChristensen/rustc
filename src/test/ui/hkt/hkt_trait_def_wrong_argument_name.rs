// HKT inspect method
// check-fail

trait Test<I<%A>> {
    fn identity(i: I<u32>) -> I<u32>;
}

impl<T> Test<Option<%B>> for Option<T> { //~ERROR: hkt argument `%B` could not be found in the definition
    fn identity(i: Option<u32>) -> Option<u32> {
        i
    }
}

fn main() {}

