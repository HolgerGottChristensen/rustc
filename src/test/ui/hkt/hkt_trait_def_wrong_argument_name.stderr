error[E9999]: hkt argument `%B` could not be found in the definition
  --> $DIR/hkt_trait_def_wrong_argument_name.rs:8:22
   |
LL | impl<T> Test<Option<%B>> for Option<T> {
   |                      ^
   |
note: expected one of the parameters `%A` inside the definition
  --> $DIR/hkt_trait_def_wrong_argument_name.rs:4:12
   |
LL | trait Test<I<%A>> {
   |            ^

error: aborting due to previous error

